src/lib.rs

```diff
@@ -326,33 +327,70 @@ macro_rules! global_wlock {
     }};
 }
 
+#[cfg(test)]
+pub fn check_test_slot_busy(slot: CK_SLOT_ID) -> bool {
+    let state = match STATE.read() {
+        Ok(r) => {
+            if !r.is_initialized() {
+                return false;
+            }
+            r
+        }
+        Err(_) => return false,
+    };
+
+    match state.get_slot(slot) {
+        Ok(_) => true,
+        Err(_) => false,
+    }
+}
+
+pub const DEFAULT_CONF_NAME: &str = "token.sql";
+
+fn find_conf() -> KResult<String> {
+    /* First check for our own env var,
+     * this has the highest precedence */
+    match env::var("KRYOPTIC_CONF") {
+        Ok(var) => return Ok(var),
+        Err(_) => (),
+    }
+    /* Freedesktop specification for data dirs first
+     * then fallback to use $HOME/.local/share, if that is also not
+     * available see if we have access to a system store */
+    let datafile = match env::var("XDG_DATA_HOME") {
+        Ok(xdg) => format!("{}/kryoptic/{}", xdg, DEFAULT_CONF_NAME),
+        Err(_) => match env::var("HOME") {
+            Ok(home) => {
+                format!("{}/.local/share/kryoptic/{}", home, DEFAULT_CONF_NAME)
+            }
+            Err(_) => format!("/var/kryoptic/public/{}", DEFAULT_CONF_NAME),
```

Comment:
should we fall back only on the presence of the HOME env variable or should we actually check the path exists before falling back?
I think that every well-behaved shell will have HOME env variable so on the first sight, I do not see the obvious case where the last case would kick in. Not sure if it is good or bad though.